File: app/javascript/helpers/time-helper.js
/**
 * helper to convert ms to time format
 *
 * @param {integer} ms
 */
export const msToTime = (ms) => {
  const time = parseFloat(ms) / 1000

  if (ms === 0) {
    return '0.000'
  }

  if (ms < 60000) {
    return `${Math.floor(time % 60)}.${Math.floor(time % 1 * 1000)}`
  }

  if (ms < 3600000) {
    return `${Math.floor(time / 60)}:${Math.floor(time % 60)}`
  }
}

File: app/helpers/time_helper.rb
module TimeHelper
  def ms_to_time(ms)
    time = ms.to_f / 1000

    return time if ms < 60000

    if ms < 3600000
      return "#{(time / 60).floor}:#{(time % 60).round(2).to_s.rjust(5, "0")}"
    end

    Time.at(time).utc.strftime("%M:%S.%L")
  end
end
